panel/src/api/request.js

```diff
@@ -33,7 +33,12 @@ export default (config) => {
       this.running++;
 
       // fetch the resquest's response
-      const response = await fetch(config.endpoint + "/" + path, options);
+      const response = await fetch(
+        [config.endpoint, path].join(
+          (config.endpoint.endsWith("/") || path.startsWith("/")) ? "" : "/"
+        ),
+        options
+      );
```

Comment:
```suggestion
    const url = config.endpoint.replace(/\/$/, "") + "/" + path.replace(/^\//, "");
    const response = await fetch(url, options);
```

What would it be like if we wrote it like this? 🤔

Reply:
@afbora I think your solution would add a slash even if no path exists. I think that's why Bastian went for the `[].join()` way initially.

Reply:
Ahh you are right.  I couldn't think it would be empty.